gradio/components/paramviewer.py

```diff
@@ -48,26 +59,20 @@ def __init__(
             render=render,
         )
 
-    def preprocess(self, payload: list[Parameter]) -> list[Parameter]:
-        """
-        Parameters:
-            payload: A list of dictionaries with keys "type", "description", and "default" for each parameter.
-        Returns:
-            A list of dictionaries with keys "type", "description", and "default" for each parameter.
-        """
+    def preprocess(self, payload: dict[str, Parameter]) -> dict[str, Parameter]:
         return payload
 
-    def postprocess(self, value: list[Parameter]) -> list[Parameter]:
-        """
-        Parameters:
-            value: A list of dictionaries with keys "type", "description", and "default" for each parameter.
-        Returns:
-            A list of dictionaries with keys "type", "description", and "default" for each parameter.
-        """
```

Comment:
We should add these docstrings back. I'm not against having this info elsewhere as well, but the docs generator pulls this information from the docstring for the pre and postprocessor return and inputs. 

This is also the correct place for this information from a maintenance PoV rather than in the class docstring.

We should also ensure docstrings in other components are consistent with this pattern, so that cloning templates gives a good experience.

Reply:
I agree that preprocess/postprocess functions are the right place for this to be documented. I could change the website to read from there instead. I think the reason we started doing it in the class docstring was because we show it on the website as one sentence.

![Screen Shot 2024-01-23 at 11 42 06 PM](https://github.com/gradio-app/gradio/assets/9021060/c92a5267-9105-46f8-bb42-f69d5d79a71b)

So its easier for the website to read this sentence from the docstrings instead of forming it from param descriptions

Reply:
Yes that sounds good, go ahead and push directly to this PR @aliabd 

I think we could write this as:

```
Supported data types:

As input: [the "return" part of the docstring of "preprocess()"]
As output: [the docstring for "value" in "postprocess()"]
```